components/brave_rewards/core/engine/hash_prefix_iterator.h

```diff
@@ -0,0 +1,115 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_REWARDS_CORE_ENGINE_HASH_PREFIX_ITERATOR_H_
+#define BRAVE_COMPONENTS_BRAVE_REWARDS_CORE_ENGINE_HASH_PREFIX_ITERATOR_H_
+
+#include <iterator>
+#include <string_view>
+
+namespace brave_rewards {
+
+// A random-access iterator over prefixes stored in an uncompressed prefix list,
+// suitable for binary search.
+class HashPrefixIterator {
```

Comment:
Hm, do we really need this thing?
If it's only to support std::binary_search, we can just compare [0.. count-1] indexes + a custom compare function instead.

Reply:
This is a rename (and cleanup) of an existing class currently used by `PrefixListReader`.

Reply:
O, thanks, I missed that part.
I see it uses to make sanity check that the first few prefixes are in order.  That code can be moved to `UpdatePrefixes`.
If we can rid off it completely, it will be great.